services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,18 +45,71 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
-  return ERR_CODE_SUCCESS;
+  if (event == NULL)
+  {
+    return ERR_CODE_INVALID_ARG; 
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, (void *) event, 0) == pdTRUE)
+  {
+    return ERR_CODE_SUCCESS;
+  }  
+
+  return ERR_CODE_QUEUE_FULL; 
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t interrupt;
```

Comment:
Good practice to initialize structs to 0s. So `thermal_mgr_event_t interrupt = {0};`. We've had bugs in the past due to structs containing garbage data.